src/main/java/org/wso2/carbon/esb/connector/auth/AmazonLambdaAuthConnector.java

```diff
@@ -0,0 +1,438 @@
+            stringToSign.append(AmazonLambdaConstants.AWS4_HMAC_SHA_256);
+            stringToSign.append(AmazonLambdaConstants.NEW_LINE);
+            stringToSign.append(amzDate);
+            stringToSign.append(AmazonLambdaConstants.NEW_LINE);
+            stringToSign.append(dateOnly);
+            stringToSign.append(AmazonLambdaConstants.FORWARD_SLASH);
+            stringToSign.append(messageContext.getProperty(AmazonLambdaConstants.REGION));
+            stringToSign.append(AmazonLambdaConstants.FORWARD_SLASH);
+            stringToSign.append(messageContext.getProperty(AmazonLambdaConstants.SERVICE));
+            stringToSign.append(AmazonLambdaConstants.FORWARD_SLASH);
+            stringToSign.append(messageContext.getProperty(AmazonLambdaConstants.TERMINATION_STRING));
+            stringToSign.append(AmazonLambdaConstants.NEW_LINE);
+            stringToSign.append(bytesToHex(hash(messageContext, canonicalRequest.toString())).toLowerCase());
+            //Creates signingKey
+            final byte[] signingKey =
+                    getSignatureKey(messageContext,
+                            messageContext.getProperty(AmazonLambdaConstants.SECRET_ACCESS_KEY).toString(),
+                            dateOnly, messageContext.getProperty(AmazonLambdaConstants.REGION).toString(),
+                            messageContext.getProperty(AmazonLambdaConstants.SERVICE).toString());
+            // Construction of authorization header value to be included in API request
+            authHeader.append(AmazonLambdaConstants.AWS4_HMAC_SHA_256);
+            authHeader.append(" ");
+            authHeader.append(AmazonLambdaConstants.CREDENTIAL);
+            authHeader.append(AmazonLambdaConstants.EQUAL);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.ACCESS_KEY_ID));
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(dateOnly);
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.REGION));
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.SERVICE));
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.TERMINATION_STRING));
+            authHeader.append(AmazonLambdaConstants.COMMA);
+            authHeader.append(AmazonLambdaConstants.SIGNED_HEADERS);
+            authHeader.append(AmazonLambdaConstants.EQUAL);
+            authHeader.append(signedHeaders);
+            authHeader.append(AmazonLambdaConstants.COMMA);
+            authHeader.append(AmazonLambdaConstants.API_SIGNATURE);
+            authHeader.append(AmazonLambdaConstants.EQUAL);
+            authHeader.append(bytesToHex(hmacSHA256(signingKey, stringToSign.toString())).toLowerCase());
```

Comment:
You can append values to authHeader like
authHeader.append().append()

Reply:
Fix with 408c7bd